### PR TITLE
Add amd measurements flow

### PR DESCRIPTION
**Pull request by rihter007**
Add PSPDisabledLegacy & PSPEnabledLegacy flows
Add detection of AMD firmware
Add parsing of MP0C2PMsg37 register

TODO: move intel registers into registers/intel folder (in another PR)

### CODE REVIEW

**Comment by rihter007** [comment]
Checked locally with the latest AMD firmware:

Set 0x -> 0x0000000000000000000000000000000000000003

Event 'pcdFirmwareVendor_measured_data': 1efb6b540c1d5540a4ad4ef4bf17b83a (*sha1.digest)
*sha1.digest(0x 0000000000000000000000000000000000000003 C42FEDAD268200CB1D15F97841C344E79DAE3320) == 0x634B3E0535CC6A07546BAD47A462ABC427BA6D28

Event 'DXE': 000000000000000000000000000000007ac07354... (len: 7397376) (*sha1.digest)
*sha1.digest(0x 634B3E0535CC6A07546BAD47A462ABC427BA6D28 F3C8D23FC5E66F863DEF8D071053B2987949BC50) == 0x40EAF1ECB357CBAAF67B52F6F6563B763464897A

Event 'separator': 00000000 (*sha1.digest)
*sha1.digest(0x 40EAF1ECB357CBAAF67B52F6F6563B763464897A 9069CA78E7450A285173431B3E52C5C25299E473) == 0x89CDFDC579DCFB24DD282AE66C73867784D18575

Resulting PCR0: 89CDFDC579DCFB24DD282AE66C73867784D18575

File: pkg/pcr/flow.go [hunk 1]
```diff
@@ -55,6 +63,12 @@ const (
 
 	// FlowIntelLegacyTXTEnabledTPM12 means a pre-CBnT flow with enabled TXT for TPM 1.2
 	FlowIntelLegacyTXTEnabledTPM12
+
+	// FlowLegacyPSPDisabled means an AMD flow with disabled PSP and with initial set of measurements
+	FlowLegacyPSPDisabled
+
+	// FlowLegacyPSPEnabled means an AMD flow with enabled PSP and with initial set of measurements
+	FlowLegacyPSPEnabled
```
Comment (xaionaro) [inline pkg/pcr/flow.go:71]: Why `Legacy`? In Intel we just used `Legacy` to emphasize that `TXT` of **pre**-CBnT flow.

Comment (rihter007) [reply pkg/pcr/flow.go:71]: Good question, because it the initial implementation of the measurements.
More measurements are expected to appear.

Comment (xaionaro) [reply pkg/pcr/flow.go:71]: OK, makes sense.

**Review by xaionaro — APPROVED** [review]

**Review by ChriMarMe — APPROVED** [review]
I have no idea what's going on, but that does not bother me to approve :D

**Pull request merged**